src/main/java/touro/tree/TriangleBoardTree.java

```diff
@@ -0,0 +1,77 @@
+package touro.tree;
+
+import touro.peg.Move;
+import touro.peg.TriangleBoard;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class TriangleBoardTree {
+    public static void main(String[] args) {
+        TriangleBoard triangleBoard = new TriangleBoard(0);
+        TriangleBoardTree triangleBoardTree = new TriangleBoardTree(triangleBoard);
+
+        System.out.println(triangleBoardTree.getSequenceToLeaf(triangleBoardTree.getLeaves().get(0)));
+        System.out.println(triangleBoardTree.getLeaves().get(0));
+
+    }
+    private List<TriangleTreeNode> leaves;
+
+    class TriangleTreeNode {
+        private TriangleBoard triangleBoard;
+        private TriangleTreeNode parent;
+        private Move move;
+        private List<TriangleTreeNode> children;
+
+        public TriangleTreeNode(TriangleBoard triangleBoard, TriangleTreeNode parent, Move move, List<TriangleTreeNode> children) {
```

Comment:
It's strange that the `TriangleTreeNode` needs a reference to the `TriangleBoard`.
Also the `children` should be an empty list because they haven't been created yet so they don't need to be passed into the constructor.

Reply:
The way we built a TriangleTreeNode was that each node is basically a board position with all its potential move outcomes, so we always pass through the constructor the current board we're dealing with (the copied board with its adjusted move) whenever we create a new node.

Reply:
Also, our node was intended to be a subtree, so that's another reason why we pass a board through it, since each tree needs a board.